course/src/main/java/org/openedx/course/presentation/unit/container/CourseUnitContainerAdapter.kt

```diff
@@ -23,27 +23,17 @@ class CourseUnitContainerAdapter(
 
     private fun unitBlockFragment(block: Block): Fragment {
         return when {
-            (block.isVideoBlock && block.studentViewData?.encodedVideos != null) -> {
+            (block.isVideoBlock &&
+                    (block.studentViewData?.encodedVideos?.videoUrl?.isEmpty() == false ||
+                            block.studentViewData?.encodedVideos?.youtube?.url?.isEmpty() == false)) -> {
```

Comment:
Please create a utility function inside `EncodedVideos` to make it more readable for checking `videoUrl` and `youtube.url`.

Reply:
done